src/mappings/mappingHandlers.ts

```diff
@@ -382,22 +448,41 @@ export async function handleNativeBalanceIncrement(event: CosmosEvent): Promise<
   }
 }
 
-async function saveNativeBalanceEvent(id: string, address: string, amount: BigInt, denom: string, event: CosmosEvent) {
-  let accountEntity = await Account.get(address)
+async function checkBalancesAccount(address: string) {
+  let accountEntity = await Account.get(address);
   if (typeof(accountEntity) === "undefined") {
     accountEntity = Account.create({id: address});
     await accountEntity.save();
   }
+}
 
+async function saveNativeBalanceEvent(id: string, address: string, amount: BigInt, denom: string, event: CosmosEvent) {
+  await checkBalancesAccount(address);
   const nativeBalanceChangeEntity = NativeBalanceChange.create({
     id,
-    accountId: address,
     balanceOffset: amount.valueOf(),
-    denom: denom,
+    denom,
+    accountId: address,
     eventId: `${messageId(event)}-${event.idx}`,
     blockId: event.block.block.id,
     transactionId: event.tx.hash,
   });
-
   await nativeBalanceChangeEntity.save()
-}
\ No newline at end of file
+}
+
+async function saveCW20BalanceEvent(id: string, address: string, amount: BigInt, contract: string, event: CosmosEvent) {
```

Comment:
nit: This function signature is quite large. After 4 or so, I tend to prefer consolidating arguments into a data structure. I think an easy way to get most of these under one object would be to use `CW20BalanceChangeEventProps`, which should be importable from the same file as `CW20BalanceChangeEvent`. This type can hold all the values that we're passing to `CW20BalanceChangeEvent.create`. You could also wrap that type in a `Partial` generic to make all of the properties optional (i.e. `Partial<CW20BalanceChangeEventProps>`).